diff -ruN Python-2.7.10/Include/timefuncs.h Python-2.7.10-new/Include/timefuncs.h
--- Python-2.7.10/Include/timefuncs.h	2015-05-23 10:09:00.000000000 -0600
+++ Python-2.7.10-new/Include/timefuncs.h	2015-06-23 07:00:03.434025300 -0600
@@ -14,10 +14,10 @@
  * to fit in a time_t.  ValueError is set on return iff the return
  * value is (time_t)-1 and PyErr_Occurred().
  */
-PyAPI_FUNC(time_t) _PyTime_DoubleToTimet(double x);
+time_t _PyTime_DoubleToTimet(double x);
 
 /* Get the current time since the epoch in seconds */
-PyAPI_FUNC(double) _PyTime_FloatTime(void);
+double _PyTime_FloatTime(void);
 
 
 #ifdef __cplusplus
diff -ruN Python-2.7.10/Modules/timemodule.c Python-2.7.10-new/Modules/timemodule.c
--- Python-2.7.10/Modules/timemodule.c	2015-05-23 10:09:21.000000000 -0600
+++ Python-2.7.10-new/Modules/timemodule.c	2015-06-23 06:59:13.591937800 -0600
@@ -1057,7 +1057,7 @@
 }
 
 /* export floattime to socketmodule.c */
-PyAPI_FUNC(double)
+double
 _PyTime_FloatTime(void)
 {
     return floattime();
