--- /home/david/Python-2.7.1/Python/pythonrun.c	2010-10-29 03:45:34.000000000 +0000
+++ Python/pythonrun.c	2011-04-23 10:45:26.488634791 +0000
@@ -73,17 +73,17 @@
 extern void _PyGILState_Fini(void);
 #endif /* WITH_THREAD */
 
-int Py_DebugFlag; /* Needed by parser.c */
-int Py_VerboseFlag; /* Needed by import.c */
-int Py_InteractiveFlag; /* Needed by Py_FdIsInteractive() below */
-int Py_InspectFlag; /* Needed to determine whether to exit at SystemError */
-int Py_NoSiteFlag; /* Suppress 'import site' */
-int Py_BytesWarningFlag; /* Warn on str(bytes) and str(buffer) */
-int Py_DontWriteBytecodeFlag; /* Suppress writing bytecode files (*.py[co]) */
+int Py_DebugFlag = 0; /* Needed by parser.c */
+int Py_VerboseFlag = 0; /* Needed by import.c */
+int Py_InteractiveFlag = 0; /* Needed by Py_FdIsInteractive() below */
+int Py_InspectFlag = 0; /* Needed to determine whether to exit at SystemError */
+int Py_NoSiteFlag = 0; /* Suppress 'import site' */
+int Py_BytesWarningFlag = 0; /* Warn on str(bytes) and str(buffer) */
+int Py_DontWriteBytecodeFlag = 0; /* Suppress writing bytecode files (*.py[co]) */
 int Py_UseClassExceptionsFlag = 1; /* Needed by bltinmodule.c: deprecated */
-int Py_FrozenFlag; /* Needed by getpath.c */
+int Py_FrozenFlag = 0; /* Needed by getpath.c */
 int Py_UnicodeFlag = 0; /* Needed by compile.c */
-int Py_IgnoreEnvironmentFlag; /* e.g. PYTHONPATH, PYTHONHOME */
+int Py_IgnoreEnvironmentFlag = 0; /* e.g. PYTHONPATH, PYTHONHOME */
 /* _XXX Py_QnewFlag should go away in 2.3.  It's true iff -Qnew is passed,
   on the command line, and is used in 2.2 by ceval.c to make all "/" divisions
   true divisions (which they will be in 2.3). */
--- /home/david/Python-2.7.1/Objects/object.c	2010-08-28 18:29:13.000000000 +0000
+++ Objects/object.c	2011-04-23 10:44:13.368316961 +0000
@@ -30,7 +30,7 @@
 #endif /* Py_REF_DEBUG */
 
 int Py_DivisionWarningFlag;
-int Py_Py3kWarningFlag;
+int Py_Py3kWarningFlag = 0;
 
 /* Object allocation routines used by NEWOBJ and NEWVAROBJ macros.
    These are used by the individual routines for object creation.
