--- a/PC/_msi.c
+++ b/PC/_msi.c
@@ -174,8 +174,8 @@
     FileTimeToLocalFileTime(&bhfi.ftLastWriteTime, &filetime);
     FileTimeToDosDateTime(&filetime, pdate, ptime);
 
-    *pattribs = (int)(bhfi.dwFileAttributes &
-        (_A_RDONLY | _A_SYSTEM | _A_HIDDEN | _A_ARCH));
+    *pattribs = (int)(bhfi.dwFileAttributes & 39);
+        /*(_A_RDONLY | _A_SYSTEM | _A_HIDDEN | _A_ARCH));*/
 
     CloseHandle(handle);
 
--- a/Modules/posixmodule.c
+++ b/Modules/posixmodule.c
@@ -414,9 +414,7 @@
 #  ifdef HAVE_DIRECT_H
 #    include <direct.h>
 #  endif
-#  ifdef HAVE_IO_H
-#    include <io.h>
-#  endif
+#  include <io.h>
 #  ifdef HAVE_PROCESS_H
 #    include <process.h>
 #  endif
