--- a/PC/_msi.c
+++ b/PC/_msi.c
@@ -174,8 +174,8 @@
     FileTimeToLocalFileTime(&bhfi.ftLastWriteTime, &filetime);
     FileTimeToDosDateTime(&filetime, pdate, ptime);
 
-    *pattribs = (int)(bhfi.dwFileAttributes &
-        (_A_RDONLY | _A_SYSTEM | _A_HIDDEN | _A_ARCH));
+    *pattribs = (int)(bhfi.dwFileAttributes & 39);
+        /*(_A_RDONLY | _A_SYSTEM | _A_HIDDEN | _A_ARCH));*/
 
     CloseHandle(handle);
 
--- a/Modules/posixmodule.c
+++ b/Modules/posixmodule.c
@@ -308,8 +308,8 @@
 #    define HAVE_PIPE       1
 #    define HAVE_SYSTEM     1
 #    define HAVE_CWAIT      1
-#    define HAVE_FSYNC      1
-#    define fsync _commit
+/*#    define HAVE_FSYNC      1*/
+/*#    define fsync _commit*/
 #  else
      /* Unix functions that the configure script doesn't check for */
 #    ifndef __VXWORKS__
