--- a/PC/_msi.c
+++ b/PC/_msi.c
@@ -174,8 +174,8 @@
     FileTimeToLocalFileTime(&bhfi.ftLastWriteTime, &filetime);
     FileTimeToDosDateTime(&filetime, pdate, ptime);
 
-    *pattribs = (int)(bhfi.dwFileAttributes &
-        (_A_RDONLY | _A_SYSTEM | _A_HIDDEN | _A_ARCH));
+    *pattribs = (int)(bhfi.dwFileAttributes & 39;
+        /*(_A_RDONLY | _A_SYSTEM | _A_HIDDEN | _A_ARCH));*/
 
     CloseHandle(handle);
